cpp/src/sampling/detail/sampling_utils_impl.cuh

```diff
@@ -285,5 +302,283 @@ sample_edges(raft::handle_t const& handle,
+                                                         edge_src_dummy_property_t{}.view(),
+                                                         edge_dst_dummy_property_t{}.view(),
+                                                         edge_weight_type_id_view,
+                                                         return_all_edges_e_op<vertex_t>{},
+                                                         do_expensive_check);
+    } else {
+      std::tie(majors, minors, weights) =
+        cugraph::extract_transform_v_frontier_outgoing_e(handle,
+                                                         graph_view,
+                                                         vertex_frontier.bucket(0),
+                                                         edge_src_dummy_property_t{}.view(),
+                                                         edge_dst_dummy_property_t{}.view(),
+                                                         *edge_weight_view,
+                                                         return_all_edges_e_op<vertex_t>{},
+                                                         do_expensive_check);
+    }
+  } else {
+    if (edge_id_type_view) {
+      std::tie(majors, minors, edge_ids, edge_types) =
+        cugraph::extract_transform_v_frontier_outgoing_e(handle,
+                                                         graph_view,
+                                                         vertex_frontier.bucket(0),
+                                                         edge_src_dummy_property_t{}.view(),
+                                                         edge_dst_dummy_property_t{}.view(),
+                                                         *edge_id_type_view,
+                                                         return_all_edges_e_op<vertex_t>{},
+                                                         do_expensive_check);
+    } else {
+      std::tie(majors, minors) =
+        cugraph::extract_transform_v_frontier_outgoing_e(handle,
+                                                         graph_view,
+                                                         vertex_frontier.bucket(0),
+                                                         edge_src_dummy_property_t{}.view(),
+                                                         edge_dst_dummy_property_t{}.view(),
+                                                         edge_dummy_property_t{}.view(),
+                                                         return_all_edges_e_op<vertex_t>{},
+                                                         do_expensive_check);
+    }
+  }
+
+  return std::make_tuple(std::move(majors),
+                         std::move(minors),
+                         std::move(edge_ids),
+                         std::move(weights),
+                         std::move(edge_types),
+                         std::move(labels));
```

Comment:
`labels` isn't updated inside `gather_one_hop_edgelist`. I suppose it will be changed later on as mentioned in the comment inside `sample_edges` ?